src/main/java/net/snowflake/client/core/SFBaseResultSet.java

```diff
@@ -113,6 +113,8 @@ public abstract class SFBaseResultSet {
 
   public abstract Object getObject(int columnIndex) throws SFException;
 
+  public abstract <T> Object getObject(int columnIndex, Class<T> object) throws SFException;
```

Comment:
this API looks strange - the result type should be T, not Object

Reply:
I agree, in reality this getObject mostly returns StructObjectWrapper which possibly could be generic. I may rename this method since it's used in higher level getObject anyway to better indicate it's purpose